### PR TITLE
CI: CMake: Separate build and test steps

### PR DESCRIPTION
**Pull request by anzz1**
Otherwise the tests may be ran and pass even if the build has errors and thus the step itself can pass with errors in build

edit: separate build and test steps for all platforms.

### CODE REVIEW

**Comment by anzz1** [comment]
For future considerations:

In GitHub actions, each row of the `run:` commands are executed separately, irregardless what the previous input returned. The step determines "success" from the error code returned by the very last run command. As `&&` means "execute if last output equals no error", `cmake --build . --config Release && ctest -C Release --output-on-failure` makes sure that the tests are only ran if the build succeeds, and that both needs to succeed for the CI step to return "success".

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -107,8 +107,7 @@ jobs:
           mkdir build
           cd build
           cmake ..
-          cmake --build . --config Release
-          ctest -C Release --output-on-failure
+          cmake --build . --config Release && ctest -C Release --output-on-failure
```
Comment (Green-Sky) [inline .github/workflows/build.yml:110]: I think it would be better for the test to be its own task instead.

**Review by Green-Sky — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (anzz1) [reply .github/workflows/build.yml]: You're absolutely right. Separated build and test phases for all platforms.

**Comment by anzz1** [comment]
Separated build and test steps as separate tasks as pointed out by @Green-Sky . Obviously should've done that in the first place, idk how I didn't think of that.

Also standardised the step id names between the platforms for 'cleanliness' sake.

**Review by Green-Sky — APPROVED** [review]
looks good.

**Pull request merged**